### PR TITLE
Handle configured links properly

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      ""
+    }
+    baseUrl + config.publicPath
+  }
+
+  private def decodeUrlFromRequest(request: HttpRequest): String = {
+    def firstHeader[T](fun: PartialFunction[HttpHeader, T]): Option[T] = request.headers.collectFirst(fun)
+
+    val protocol: String = firstHeader {
+      case `X-Forwarded-Proto`(forwardedProto) => forwardedProto
```
Comment (lciolecki) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala:41]: Header names as const?

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/config/ComponentsActionConfigExtractor.scala [hunk 2]
```diff
@@ -5,20 +5,21 @@ import net.ceedubs.ficus.readers.{OptionReader, ValueReader}
...
   import ComponentLinkConfig._
 
   def isAvailable(componentType: ComponentType): Boolean = supportedComponentTypes.isEmpty || supportedComponentTypes.exists(_.contains(componentType))
 
   def toComponentLink(componentId: ComponentId, componentName: String): ComponentLink = ComponentLink(
       id,
       fillByComponentData(title, componentId, componentName),
-      URI.create(fillByComponentData(icon.toString, componentId, componentName, urlOption = true)),
-      new URL(fillByComponentData(url.toString, componentId, componentName, urlOption = true))
+      NuIcon(URI.create(fillByComponentData(icon, componentId, componentName, urlOption = true))),
```
Comment (lciolecki) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/config/ComponentsActionConfigExtractor.scala:21]: Right now you don't nee URI.create

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessToolbarService.scala [hunk 3]
```diff
@@ -79,14 +86,15 @@ object ToolbarButton {
     config.`type`,
     config.name.map(t => fillByProcessData(t, process)),
     config.title.map(t => fillByProcessData(t, process)),
-    config.icon.map(i => fillByProcessData(i, process, urlOption = true)),
-    config.url.map(th => fillByProcessData(th, process, urlOption = true)),
+    config.icon.map(i => NuIcon(URI.create(fillByProcessData(i, process, urlOption = true)))),
```
Comment (lciolecki) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessToolbarService.scala:89]: Right now you don't need URI.create

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import akka.http.scaladsl.model.{HttpHeader, HttpRequest}
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.server.Directives._
+import io.circe.Encoder
+import pl.touk.nussknacker.restmodel.NuLink
+
+import java.net.URL
+import scala.util.Try
+
+case class LinkEncodingConfig(publicPath: String = "/", encodeAbsoluteUrl: Boolean = true)
```
Comment (lciolecki) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala:13]: Maybe remove this defaults and create object with default implementation?

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    val port = firstHeader {
+      case header if header.name().toLowerCase == xForwardedPort => Try(header.value().toInt).toOption
+    }.flatten.getOrElse(authority.port)
+
+    val path = firstHeader {
+      case header if header.name().toLowerCase == xForwardedPath => header.value()
+    }.getOrElse("")
+
+    val normalized = authority.copy(host = host, port = port).normalizedFor(protocol)
+    val normalizedPort = if (normalized.port == 0) -1 else normalized.port
```
Comment (lciolecki) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NuLinkEncoder.scala:59]: What does mean port 0? Let's create some const for it.

**Comment by mproch** [comment]
Closing for now, we don't have enough resources to implement it properly now...

**Pull request closed without merging**